_posts/2023-05-10-group-equivariant-cnn-3/group-equivariant-cnn-3.Rmd

```diff
@@ -0,0 +1,492 @@
+
+[^1]: This is nicely explained in, for example, Jakob Schwichtenberg's [Physics from symmetry](http://physicsfromsymmetry.com/).
+
+## Purpose and scope of this post
+
+Here, I introduce [`escnn`](https://github.com/QUVA-Lab/escnn), a PyTorch extension that implements forms of group equivariance for CNNs operating on the plane or in (3d) space. The library is used in various, amply illustrated research papers; it is appropriately documented; and it comes with introductory notebooks both relating the math and exercising the code. Why, then, not just refer to the [first notebook](https://github.com/QUVA-Lab/escnn/blob/master/examples/introduction.ipynb), and immediately start using it for some experiment?
+
+In fact, this post should -- as quite a few texts I've written -- be regarded as an introduction to an introduction. To me, this topic seems anything but easy, for various reasons. Of course, there's the math. But as so often in machine learning, you don't need to go to great depths to be able to apply an algorithm correctly. So if not the math itself, what generates the difficulty? For me, it's two things.
+
+First, to map my understanding of the mathematical concepts to the terminology used in the library, and from there, to correct use and application. Expressed schematically: We have a concept A, which figures (among other concepts) in technical term (or object class) B. What does my understanding of A tell me about how object class B is to be used correctly? More importantly: How do I use it to best attain my goal C? -- This first difficulty I'll address in a very pragmatic way. I'll neither dwell on mathematical details, nor try to establish the links between A, B, and C in detail. Instead, I'll present the characters[^2] in this story by asking what they're good for.
```

Comment:
```suggestion
First, to map my understanding of the mathematical concepts to the terminology used in the library, and from there, to correct use and application. Expressed schematically: We have a concept A, which figures (among other concepts) in technical term (or object class) B. What does my understanding of A tell me about how object class B is to be used correctly? More importantly: How do I use it to best attain my goal C? This first difficulty I'll address in a very pragmatic way. I'll neither dwell on mathematical details, nor try to establish the links between A, B, and C in detail. Instead, I'll present the characters[^2] in this story by asking what they're good for.
```

Reply:
I meant it in sort of a more general way... Certainly, what's included are the library objects (B), but a broader understanding where the math terms (A), too, are characters of the story, works fine as well...

I think that overall in this post, I'm playing "fast and loose", so to say, with the writing style, trying to combine maximal precision and entertainment (insofar that is possible with such a topic)...